plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -0,0 +1,520 @@
+                                        GetTablesRequest::setNextToken,
+                                        GetTablesResult::getNextToken)
+                                        .map(GetTablesResult::getTableList)
+                                        .flatMap(List::stream)
+                                        .map(table -> new SchemaTableName(glueNamespace, table.getName()));
+                            }
+                            catch (EntityNotFoundException e) {
+                                return Stream.empty();
+                            }
+                        })
+                        .collect(toImmutableList());
+            });
+        }
+        catch (AmazonServiceException e) {
+            throw new TrinoException(ICEBERG_CATALOG_ERROR, e);
+        }
+    }
+
+    @Override
+    public Table loadTable(ConnectorSession session, SchemaTableName table)
+    {
+        TableMetadata metadata = tableMetadataCache.computeIfAbsent(
+                table,
+                ignore -> {
+                    TableOperations operations = tableOperationsProvider.createTableOperations(
+                            this,
+                            session,
+                            table.getSchemaName(),
+                            table.getTableName(),
+                            Optional.empty(),
+                            Optional.empty());
+                    return new BaseTable(operations, quotedTableName(table)).operations().current();
+                });
+
+        IcebergTableOperations operations = tableOperationsProvider.createTableOperations(
+                this,
+                session,
+                table.getSchemaName(),
+                table.getTableName(),
+                Optional.empty(),
+                Optional.empty());
+        operations.initializeFromMetadata(metadata);
+        return new BaseTable(operations, quotedTableName(table));
+    }
+
+    @Override
+    public boolean dropTable(ConnectorSession session, SchemaTableName schemaTableName, boolean purgeData)
+    {
+        Table table = loadTable(session, schemaTableName);
+        validateTableCanBeDropped(table, schemaTableName);
+        try {
+            deleteTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+        }
+        catch (AmazonServiceException e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, e);
+        }
+
+        Path tableLocation = new Path(table.location());
+        if (purgeData) {
+            try {
+                hdfsEnvironment.getFileSystem(new HdfsEnvironment.HdfsContext(session), tableLocation).delete(tableLocation, true);
+            }
+            catch (Exception e) {
+                // don't fail if unable to delete path
+                log.warn(e, "Failed to delete path: " + tableLocation);
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public Transaction newCreateTableTransaction(
```

Comment:
Looks identical to `io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog#newCreateTableTransaction`.
should this be unified?